File: Ejercicio_03/Domain/Entities/Persona.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{
    public class Persona
    {
        #region atributos
        private int _id;
        private string _nombre;
        private string _apellidos;
        private string _telefono;
        private string _direccion;
        private string _foto;
        private DateOnly _fechaNacimiento;
        private int _idDepartamento;
        #endregion

        #region constructores
        public Persona() { }
        public Persona(int id, string nombre, string apellidos, string telefono, string direccion, string foto, DateOnly fechaNacimiento, int idDepartamento)
        {
            _id = id;
            _nombre = nombre;
            _apellidos = apellidos;
            _telefono = telefono;
            _direccion = direccion;
            _foto = foto;
            _fechaNacimiento = fechaNacimiento;
            _idDepartamento = idDepartamento;
        }
        #endregion

        #region propiedades
        [Required(ErrorMessage = "Campo obligatorio")]
        public int id
        {
            get { return _id; }
            set { _id = value; } // Se lo he puesto porque cuando estoy obteniendo los datos de la BD tengo que actualizarlo
        }
        [Required(ErrorMessage = "Campo obligatorio")]
        [MaxLength(30, ErrorMessage = "Máximo 30 caracteres")]
        public string nombre
        {
            get { return _nombre; }
            set { _nombre = value; }
        }
        [Required(ErrorMessage = "Campo obligatorio")]
        [MaxLength(60, ErrorMessage = "Máximo 60 caracteres")]
        public string apellidos
        {
            get { return _apellidos; }
            set { _apellidos = value; }
        }
        [StringLength(9, MinimumLength = 9, ErrorMessage = "Debe tener exactamente 9 números")]
        public string telefono
        {
            get { return _telefono; }
            set { _telefono = value; }
        }
        public string direccion
        {
            get { return _direccion; }
            set { _direccion = value; }
        }
        public string foto
        {
            get { return _foto; }
            set { _foto = value; }
        }
        [DataType(DataType.Date)]
        public DateOnly fechaNacimiento
        {
            get { return _fechaNacimiento; }
            set { _fechaNacimiento = value; }
        }
        public int idDepartamento
        {
            get { return _idDepartamento; }
            set { _idDepartamento = value; }
        }
        #endregion
    }
}

File: Angular/CRUD/src/domain/entities/Persona.ts
export class Persona {

    private _id: number
	private _nombre: string
	private _apellidos: string
	private _telefono: string
	private _direccion: string
	private _foto: string
	private _fechaNacimiento: Date
	private _idDepartamento: number

    constructor(id:number, nombre: string, apellidos: string, telefono: string, direccion: string, foto: string, fechaNacimiento: Date, idDepartamento: number) {
        this._id = id
        this._nombre = nombre
        this._apellidos = apellidos
        this._telefono = telefono
        this._direccion = direccion
        this._foto = foto
        this._fechaNacimiento = fechaNacimiento
        this._idDepartamento = idDepartamento
    }

    public get id(): number {
        return this._id
    }

    public set id(id: number) {
        this._id = id
    }

    public get nombre(): string {
        return this._nombre
    }

    public set nombre(nombre: string) {
        this._nombre = nombre
    }

    public get apellidos(): string {
        return this._apellidos
    }

    public set apellidos(apellidos: string) {
        this._apellidos = apellidos
    }

    public get telefono(): string {
        return this._telefono
    }

    public set telefono(telefono: string) {
        this._telefono = telefono
    }

    public get direccion(): string {
        return this._direccion
    }

    public set direccion(direccion: string) {
        this._direccion = direccion
    }

    public get foto(): string {
        return this._foto
    }

    public set foto(foto: string) {
        this._foto = foto
    }

    public get fechaNacimiento(): Date {
        return this._fechaNacimiento
    }

    public set fechaNacimiento(fechaNacimiento: Date) {
        this._fechaNacimiento = fechaNacimiento
    }

    public get idDepartamento(): number {
        return this._idDepartamento
    }

    public set idDepartamento(idDepartamento: number) {
        this._idDepartamento = idDepartamento
    }
}